esmvalcore/preprocessor/_time.py

```diff
@@ -246,6 +296,213 @@ def spans_three_months(time):
     return cube.extract(three_months_bound)
 
 
+def annual_statistics(cube, operator='mean'):
+    """
+    Compute annual statistics.
+
+    Note that this function does not weight the annual mean if
+    uneven time periods are present. Ie, all data inside the year
+    are treated equally.
+
+    Parameters
+    ----------
+    cube: iris.cube.Cube
+        input cube.
+
+    operator: str, optional
+        Select operator to apply.
+        Available operators: 'mean', 'median', 'std_dev', 'variance', 'min',
+        'max'
+
+    Returns
+    -------
+    iris.cube.Cube
+        Annual mean cube
+    """
+    # time_weights = get_time_weights(cube)
+
+    # TODO: Add weighting in time dimension. See iris issue 3290
+    # https://github.com/SciTools/iris/issues/3290
+
+    operator = get_iris_analysis_operation(operator)
+
+    if not cube.coords('year'):
+        iris.coord_categorisation.add_year(cube, 'time')
+    return cube.aggregated_by('year', operator)
+
+
+def decadal_statistics(cube, operator='mean'):
+    """
+    Compute decadal statistics.
+
+    Note that this function does not weight the decadal mean if
+    uneven time periods are present. Ie, all data inside the decade
+    are treated equally.
+
+    Parameters
+    ----------
+    cube: iris.cube.Cube
+        input cube.
+
+    operator: str, optional
+        Select operator to apply.
+        Available operators: 'mean', 'median', 'std_dev', 'variance', 'min',
+        'max'
+
+    Returns
+    -------
+    iris.cube.Cube
+        Annual mean cube
+    """
+    # time_weights = get_time_weights(cube)
+
+    # TODO: Add weighting in time dimension. See iris issue 3290
+    # https://github.com/SciTools/iris/issues/3290
+
+    operator = get_iris_analysis_operation(operator)
+
+    if not cube.coords('decade'):
+        def get_decade(coord, value):
+            """Callback function to get decades from cube."""
+            date = coord.units.num2date(value)
+            return date.year - date.year % 10
+
+        iris.coord_categorisation.add_categorised_coord(
+            cube, 'decade', 'time', get_decade)
+
+    return cube.aggregated_by('decade', operator)
+
+
+def climate_statistics(cube, operator='mean', period='full'):
+    """
+    Compute climate statistics with the specified granularity.
+
+    Computes daily, monthly, seasonal or yearly statistics for the
+    full available period
+
+    Parameters
+    ----------
+    cube: iris.cube.Cube
+        input cube.
+
+    operator: str, optional
+        Select operator to apply.
+        Available operators: 'mean', 'median', 'std_dev', 'variance', 'min',
+        'max'
+
+    period: str, optional
+        Period to compute the statistic over.
+        Available periods: 'full', 'season', 'seasonal', 'monthly', 'month',
```

Comment:
Maybe it would be better to just support a single name per period. This makes recipes easier to understand and reduces the amount of code.

Reply:
I added the most common ways of naming the frequencies to avoid users searching for the correct spelling